Ported std/var to ReductionOpInfo and minimum/maximum to BinaryUfuncInfo

heitorschueroff opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63978

Differential Revision: [D30558877](https://our.internmc.facebook.com/intern/diff/D30558877)

heitorschueroff committed:
Ported std/var to ReductionOpInfo and minimum/maximum to BinaryUfuncInfo

[ghstack-poisoned]

heitorschueroff committed:
Update on "Ported std/var to ReductionOpInfo and minimum/maximum to BinaryUfuncInfo"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Ported std/var to ReductionOpInfo and minimum/maximum to BinaryUfuncInfo"


Differential Revision: [D30558877](https://our.internmc.facebook.com/intern/diff/D30558877)

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -7064,16 +7079,56 @@ def wrapper(x: np.ndarray, *args, **kwargs):
-           dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
-           supports_forward_ad=True,
-           sample_inputs_func=sample_inputs_max_min_binary,),
+    BinaryUfuncInfo(
```
Neat

mruberry reviewed (approved):
Cool! Great to have these ported and their limitations/oddities so clearly documented so we can understand how to fix them!

heitorschueroff committed:
Update on "Ported std/var to ReductionOpInfo and minimum/maximum to BinaryUfuncInfo"


Differential Revision: [D30558877](https://our.internmc.facebook.com/intern/diff/D30558877)

[ghstack-poisoned]

heitorschueroff commented:
@heitorschueroff has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30558877).
